scalikejdbc-streams/src/main/scala/scalikejdbc/streams/ExtractedResultIterator.scala

```diff
@@ -0,0 +1,55 @@
+package scalikejdbc.streams
+
+import java.sql.ResultSet
+
+import scalikejdbc.{ ResultSetCursor, WrappedResultSet }
+
+abstract class ExtractedResultIterator[+A](rs: ResultSet, autoClose: Boolean)(extract: WrappedResultSet => A) extends BufferedIterator[A] with CloseableIterator[A] {
```

Comment:
I prefer `ResultSetExtractionIterator` because the name should explain actually `ResultSet` is underlying and this class is closeable about it.